tasks/main.yml

```diff
@@ -94,18 +99,20 @@
     protocol: "{{ purefa_syslog.protocol }}"
     fa_url: "{{ purefa_url }}"
     api_token: "{{ purefa_api_token }}"
+  when: purefa_syslog is defined
   tags:
   - purefa_syslog
 
 - name: Set SMTP settings
   purefa_smtp:
-    state: "{{ purefa_smtp.state | default('present')}}"
+    state: "{{ 'present' if purefa_smtp else 'absent' }}"
```

Comment:
Is the if/else needed? if `purefa_smtp` isn't defined, this task won't run anyway, due to the `when` on line 115

Reply:
It is needed because the user might want to define the variable `purefa_smtp` as empty and therefore clear out the current values.

Reply:
Ah I see. Makes sense.